docs/pages/docs/routing/middleware.mdx

```diff
@@ -452,82 +454,51 @@ export async function updateSession(
           return request.cookies.getAll();
         },
         setAll(cookiesToSet) {
-          cookiesToSet.forEach(({ name, value }) =>
-            request.cookies.set(name, value)
+          cookiesToSet.forEach(({name, value, options}) =>
+            request.cookies.set(name, value, options)
           );
-          response = NextResponse.next({
-            request,
-          });
```

Comment:
Although it makes sense, it is recommended in the Supabase docs. This whole setup is currently working in a project of mine. I'm kind of hesitant of removing things from docs :sweat_smile:

Reply:
Right, but the major difference with your adapted code is that in `utils/supabase/middleware.ts` we now receive a response and don't have to create one, right? Logically it would make sense to me to no longer create one.

I'm honestly not sure under what circumstances the previous implementation worked, my assumption would be that the `response` from `next-intl` needs to be returned eventually from the middleware.

Can you by chance verify my proposed change here in your app?

Reply:
I can confirm everything seems to work just fine with the removal you proposed.

Honestly, I don't understand what this `setAll` piece of code does, as it all looks very redundant. We can basically also remove the piece above it, where it sets the cookies on the request, I think?